src/pkg/services/m365/api/mail.go

```diff
@@ -165,6 +165,55 @@ func (c Mail) GetContainerByID(
 	return c.GetFolder(ctx, userID, containerID)
 }
 
+// GetContainerByName fetches a folder by name
+func (c Mail) GetContainerByName(
+	ctx context.Context,
+	userID, containerName string,
+) (graph.Container, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", containerName)
+	options := &users.ItemMailFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemMailFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
+	ctx = clues.Add(ctx, "container_name", containerName)
+
+	resp, err := c.Stable.
+		Client().
+		Users().
+		ByUserId(userID).
+		MailFolders().
+		Get(ctx, options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err).WithClues(ctx)
+	}
+
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("container not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one container with the provided name.
+	// Return an error if multiple container exist (unlikely) or if no container
+	// is found.
+	if len(gv) != 1 {
```

Comment:
given that mail allows a complete folder hierarchy this doesn't seem like a reasonable assumption to make

Reply:
Since the request is non-recursive and only includes top level folders (and mailfolders cannot have duplicate names), this is safe.  I'll add a comment explaining those assumptions.